src/App.tsx

```diff
@@ -15,49 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((prev, curr) => {
+    if (sortType === SortType.ALPHABET) {
+      return prev.localeCompare(curr);
+      // eslint-disable-next-line
+    } else if (sortType === SortType.LENGTH) {
+      return prev.length - curr.length;
+    }
+
+    return 0;
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
+export const App: React.FC<{}> = () => {
+  const [isReversed, reverseList] = useState(false);
+  const [sortType, sortItems] = useState(SortType.NONE);
+
+  const goods = getReorderedGoods(goodsFromServer, { isReversed, sortType });
+  const activeRemoveButton = isReversed || sortType !== SortType.NONE;
```

Comment:
great job saving the checking conditions in a variable!

Reply:
🔥 🔥 🔥 
`isRemoveButtonActive` would be better since it is a boolean value